src/Server/TCPHandler.cpp

```diff
@@ -483,6 +491,9 @@ void TCPHandler::runImpl()
         {
             if (exception)
             {
+                if (thread_trace_context)
+                    thread_trace_context->root_span.addAttribute(*exception);
```

Comment:
I suppose functions `addAttribute(const Exception&)` and `addAttribute(std::exception_ptr)` should be made noexcept, i.e. have an internal try-catch which swallows exceptions during push_back into the attribute map. Why? Here, everything is exception-safe, but if you look into the usage of both methods in file "ThreadPool.cpp", you will find that an exception during OTel tracing will skip the exception handler in the scheduling system. If we have a choice for a malfunction in OpenTelemetry (= missing attribute) or a malfunction in scheduling, I would rather choose the former.

Reply:
Fixed